opendatahub-io/odh-dashboard
# [RHOAIENG-2981] Add Run Archive tab & restore ability

Closes: [RHOAIENG-2981](https://issues.redhat.com/browse/RHOAIENG-2981)

## Description
- Created an Archived runs tab
- Renamed "triggered" to "active" and renamed "scheduled" to "schedules"
- Added default values fed from search params for the "Create schedule" page to distinguish from "Create run" (although using the same component still)
- Added API functions for updating a run to be marked as archived or unarchived, fetching archived runs and active runs

**Notable components added:**
- Tab for Archive runs
- Archive run modal
- Bulk Archive runs modal
- Restore run modal
- Bulk Restore runs modal

## How Has This Been Tested?
Updated existing tests to verify deletion and creation still. Restore and archive tests will be added separately.

## How to test
1. Create a non-scheduled run (an active run), verify the table's primary and secondary toolbar actions, and all actions for the individual row, including the Archive action from the row.
2. Once archived, navigate to the Archive tab, verify the run exists, verify the table's primary and secondary toolbar actions, and all actions for the individual row, including the Restore action from the row.
3. After restored, navigate back to the Active runs tab, create a 2nd active run, select all checkboxes, try to archive multiple, verify the bulk Archive modal and verify those runs exist in the Archived runs table afterwards.
4. Verify restoring multiple from the Archived runs tab
5. Verify deleting multiple from the Archived runs tab

### Demos (cc @yannnz)
#### Archive single run
https://github.com/opendatahub-io/odh-dashboard/assets/96431149/52aab3df-2c2a-4fd3-842c-6eb95469e1e4
#### Restore single run
https://github.com/opendatahub-io/odh-dashboard/assets/96431149/230de3f5-3591-4328-9afc-971cb63812f7
#### Archive multiple runs
https://github.com/opendatahub-io/odh-dashboard/assets/96431149/cc45a14f-8acd-4525-ac2f-a65220ad4e37
#### Restore multiple runs
https://github.com/opendatahub-io/odh-dashboard/assets/96431149/84aa5566-1f8e-452c-89d6-7c5283498014
#### Create schedule with default values
https://github.com/opendatahub-io/odh-dashboard/assets/96431149/120e2a51-0ee7-4d7d-914b-1cd2e4b0dfe3

## Request review criteria:
Self checklist (all need to be checked):

If you have UI changes: 

After the PR is posted & before it merges:
